docs/notebook_cleaner.py

```diff
@@ -0,0 +1,86 @@
+import click
+import json
+import os
+
+DOCS_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "source")
+
+
+def _get_ipython_notebooks(docs_source):
+    directories_to_skip = ["_templates", "generated", ".ipynb_checkpoints"]
+    notebooks = []
+    for root, _, filenames in os.walk(docs_source):
+        if any(dir_ in root for dir_ in directories_to_skip):
+            continue
+        for filename in filenames:
+            if filename.endswith(".ipynb"):
+                notebooks.append(os.path.join(root, filename))
+    return notebooks
+
+
+def _check_execution_and_output(notebook):
```

Comment:
This function name mentions outputs, but I don't see that we are checking outputs anywhere. Is it ever possible to get something in the cell outputs with an execution count of None?

Reply:
Good call! I don't think it would be possible unless a user goes in and edits the notebooks via text-editor. I can add a check for outputs, however